applications/FluidDynamicsApplication/custom_conditions/fs_werner_wengle_wall_condition.h

```diff
@@ -242,87 +242,52 @@ class FSWernerWengleWallCondition: public Condition
 	{
 		KRATOS_TRY;
 
-		const array_1d<double,3>& rNormal = this->GetValue(NORMAL);
-		if (norm_2(rNormal) == 0.0)
-		  {
-		    std::cout << "error on condition -> " << this->Id() << std::endl;
-		    KRATOS_THROW_ERROR(std::logic_error, "NORMAL must be calculated before using this condition","");
-		  }
+		if (this->Is(SLIP))
+		{
+			const array_1d<double, 3> &rNormal = this->GetValue(NORMAL);
```

Comment:
I would add a check in `Check` if `NORMAL` has been assigned
this way you will notice right away if you didn't compute the normals